app/controllers/posts_controller.rb

```diff
@@ -22,12 +22,13 @@ def create
 
   def index
     @posts = Post.user_post(current_user).order(created_at: :desc)
+    @user_likes = current_user.likes.where(post_id: @posts.map(&:id)).index_by(&:post_id)
   end
 
   def edit
-    return if current_user == @post.user
-
-    redirect_to root_path, alert: 'You are not authorized to edit this post.'
+    unless current_user == @post.user
+      redirect_to root_path, alert: 'You are not authorized to edit this post.'
+    end
   end
 
   def show
```

Comment:
```suggestion
  def show; end
```

Reply:
this is not necessary any more `@post = Post.find(params[:id])` because do you set the `set_post` in the private area